File: Concesionario/src/gestorAplicacion/personal/Mecanico.java
package gestorAplicacion.personal;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Scanner;
import gestorAplicacion.activos.*;

public class Mecanico extends Trabajador implements Serializable{
	private static final long serialVersionUID = 1L;
	static ArrayList<Mecanico> mecanicos = new ArrayList<Mecanico>();
	String autos;
	String especialidad;
	public long pagoSvcs=0;
	long manoObra;
	ArrayList<String> horario=new ArrayList<String>() {{add("9:00-11:00");add("11:00-1:00");add("2:00-4:00");add("4:00-6:00");}};

	public Mecanico(String nombre, long cedula, long telefono, String correo, String direccion, double salario,
			String banco, long cuentaBanco, String autos,String especialidad,long manoObra) {
		super(nombre, cedula, telefono, correo, direccion, salario, banco, cuentaBanco);
		this.autos = autos;
		this.manoObra=manoObra;
		this.especialidad=especialidad;
		Mecanico.mecanicos.add(this);
	}
    public ArrayList<String> getHorario() {
        return horario;
    }

    public void setHorario(ArrayList<String> horario) {
        this.horario = horario;
    }
    public long  getManoObra() {
        return manoObra;
    }

    public void setManoObra(long manoObra) {
        this.manoObra = manoObra;
    }

	public String getAutos() {
		return autos;
	}

	public void setAutos(String autos) {
		this.autos = autos;
	}
	public void setEspecialidad(String especialidad) {
		this.especialidad=especialidad;
	}
	public String getEspecialidad() {
		return especialidad;
	}
	
	public static ArrayList<Mecanico> getMecanicos() {
		return mecanicos;
	}
	
	public static void setMecanicos(ArrayList<Mecanico> mecanicos) {
		Mecanico.mecanicos = mecanicos;
	}
	public static ArrayList<Mecanico> mecanicoDisponible(Auto auto) {
	    Scanner sc = new Scanner(System.in);
	    byte input;
	    String salir = null;
	    String especialidad=null;
	    ArrayList<Mecanico> mechs = new ArrayList<Mecanico>();
	    
	    do {

	        input = sc.nextByte();

	        switch (input) {
	        case 1:
                especialidad="Pintura";
        	    for (Mecanico mecanico : getMecanicos()) {
        	        if (especialidad.equals(mecanico.getEspecialidad())&& auto.getMarca().equals(mecanico.getAutos())&&mecanico.getHorario().size()>=1) {
        	            mechs.add(mecanico);
        	        }
        	        salir="si";
        	    }
                break;
            case 2:
                especialidad="Llantas";
        	    for (Mecanico mecanico : getMecanicos()) {
        	    	if (especialidad.equals(mecanico.getEspecialidad())&& auto.getMarca().equals(mecanico.getAutos())) {
        	            mechs.add(mecanico);
        	        }
        	        salir="si";
        	    }
                break;
            case 3:
                especialidad="Motor";
        	    for (Mecanico mecanico : getMecanicos()) {
        	    	if (especialidad.equals(mecanico.getEspecialidad())&& auto.getMarca().equals(mecanico.getAutos())) {
        	            mechs.add(mecanico);
        	        }
        	        salir="si";
        	    }
                break;
            case 4:
                especialidad="Frenos";
        	    for (Mecanico mecanico : getMecanicos()) {
        	    	if (especialidad.equals(mecanico.getEspecialidad())&& auto.getMarca().equals(mecanico.getAutos())) {
        	            mechs.add(mecanico);
        	        }
        	        salir="si";
        	    }
                break;
            case 5:
                especialidad="ModificacionPintura";
        	    for (Mecanico mecanico : getMecanicos()) {
        	    	if (especialidad.equals(mecanico.getEspecialidad())&& auto.getMarca().equals(mecanico.getAutos())) {
        	            mechs.add(mecanico);
        	        }
        	        salir="si";
        	    }
                break;
                
            case 6:
                especialidad="ModificacionLlantas";
        	    for (Mecanico mecanico : getMecanicos()) {
        	    	if (especialidad.equals(mecanico.getEspecialidad())&& auto.getMarca().equals(mecanico.getAutos())) {
        	            mechs.add(mecanico);
        	        }
        	        salir="si";
        	    }
                break;
                
            case 7:
                especialidad="ModificacionSonido";
        	    for (Mecanico mecanico : getMecanicos()) {
        	    	if (especialidad.equals(mecanico.getEspecialidad())&& auto.getMarca().equals(mecanico.getAutos())) {
        	            mechs.add(mecanico);
        	        }
        	        salir="si";
        	    }
                break;
                
            case 8:
                especialidad="ModificacionFrenos";
        	    for (Mecanico mecanico : getMecanicos()) {
        	    	if (especialidad.equals(mecanico.getEspecialidad())&& auto.getMarca().equals(mecanico.getAutos())) {
        	            mechs.add(mecanico);
        	        }
        	        salir="si";
        	    }
                break;
                
            case 9:
                especialidad="ModificacionEscape";
        	    for (Mecanico mecanico : getMecanicos()) {
        	    	if (especialidad.equals(mecanico.getEspecialidad())&& auto.getMarca().equals(mecanico.getAutos())) {
        	            mechs.add(mecanico);
        	        }
        	        salir="si";
        	    }
                break;
			default:
				System.out.print("\n¿Salir? (si/no)");
				salir = sc.nextLine();
			} 
	    } while(salir.equals(null));
	    return mechs;
	}
	public static Mecanico getMecanicoPorCedula(long cedula){
		Mecanico finder = null;
		for(Mecanico mecanico: mecanicos) {
			if(mecanico.getCedula()==cedula) {
				finder = mecanico;
			} 
		
		}return finder;
	}
	public static byte readByte() {
	    Scanner scanner = new Scanner(System.in);
	    byte num = scanner.nextByte();
	    return num;
	}
	@Override
	public String getNombre() {
		return nombre;
	}

	@Override
	public long getCedula() {
		// TODO Auto-generated method stub
		return cedula;
	}

	@Override
	public long getTelefono() {
		// TODO Auto-generated method stub
		return telefono;
	}

	@Override
	public String getCorreo() {
		// TODO Auto-generated method stub
		return correo;
	}

	@Override
	public String getDireccion() {
		// TODO Auto-generated method stub
		return direccion;
	}

	@Override
	public void setNombre(String nombre) {
		this.nombre=nombre;
	}

	@Override
	public void setCedula(long cedula) {
		this.cedula=cedula;
	}

	@Override
	public void setTelefono(long telefono) {
		this.telefono=telefono;
	}

	@Override
	public void setCorreo(String correo) {
		this.correo=correo;
	}

	@Override
	public void setDireccion(String direccion) {
		this.direccion=direccion;
	}

	@Override
	int calcularSalario() {
		return (int) (pagoSvcs+getSalario());
	}
	@Override
	public String info() {
		String texto = "Nombre del Mecanico: " + getNombre() + "\n";
		return texto;
	}
	
}

File: PYTHON/Concesionario/gestorAplicacion/Personal/mecanico.py
import sys
import os
ruta_activos = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'gestorAplicacion', 'Activos'))
ruta_personal = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'gestorAplicacion', 'Personal'))
ruta_gestor = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
sys.path.append(ruta_activos)
sys.path.append(ruta_personal)
sys.path.append(ruta_gestor)


from trabajador import Trabajador

class Mecanico(Trabajador):
    mecanicos = []
    horario = []

    def __init__(self, nombre, cedula, telefono, correo, direccion, salario, banco, cuentaBanco, autos, especialidad, manoObra):
        super().__init__(nombre, cedula, telefono, correo, direccion, salario, banco, cuentaBanco)
        self.autos = autos
        self.especialidad = especialidad
        self.pagoSvcs = 0
        self.manoObra = manoObra
        Mecanico.mecanicos.add(self)
        self.horario.append("9:00-11:00")
        self.horario.append("11:00-1:00")
        self.horario.append("2:00-4:00")
        self.horario.append("4:00-6:00")

    def get_horario(self):
        return self.horario

    def set_horario(self, horario):
        self.horario = horario

    def get_manoObra(self):
        return self.manoObra

    def set_manoObra(self, manoObra):
        self.manoObra = manoObra

    def get_autos(self):
        return self.autos

    def set_autos(self, autos):
        self.autos = autos

    def setEspecialidad(self, especialidad):
        self.especialidad = especialidad

    def get_especialidad(self):
        return self.especialidad

    @staticmethod
    def get_mecanicos():
        return Mecanico.mecanicos

    @staticmethod
    def set_mecanicos(mecanicos):
        Mecanico.mecanicos = mecanicos

    @staticmethod
    def mecanicoDisponible(auto):
        input = 0
        salir = None
        especialidad = None
        mechs = []

        while salir is None:
            input = int(input())

            if input == 1:
                especialidad = "Pintura"
                for mecanico in Mecanico.get_mecanicos():
                    if especialidad == mecanico.get_especialidad() and auto.get_marca() == mecanico.get_autos() and len(mecanico.get_horario()) >= 1:
                        mechs.add(mecanico)
                    salir = "si"
            elif input == 2:
                especialidad = "Llantas"
                for mecanico in Mecanico.get_mecanicos():
                    if especialidad == mecanico.get_especialidad() and auto.get_marca() == mecanico.get_autos():
                        mechs.add(mecanico)
                    salir = "si"
            elif input == 3:
                especialidad = "Motor"
                for mecanico in Mecanico.get_mecanicos():
                    if especialidad == mecanico.get_especialidad() and auto.get_marca() == mecanico.get_autos():
                        mechs.add(mecanico)
                    salir = "si"
            elif input == 4:
                especialidad = "Frenos"
                for mecanico in Mecanico.get_mecanicos():
                    if especialidad == mecanico.get_especialidad() and auto.get_marca() == mecanico.get_autos():
                        mechs.add(mecanico)
                    salir = "si"
            elif input == 5:
                especialidad = "ModificacionPintura"
                for mecanico in Mecanico.get_mecanicos():
                    if especialidad == mecanico.get_especialidad() and auto.get_marca() == mecanico.get_autos():
                        mechs.add(mecanico)
                    salir = "si"
            elif input == 6:
                especialidad = "ModificacionLlantas"
                for mecanico in Mecanico.get_mecanicos():
                    if especialidad == mecanico.get_especialidad() and auto.get_marca() == mecanico.get_autos():
                        mechs.add(mecanico)
                    salir = "si"
            elif input == 7:
                especialidad = "ModificacionSonido"
                for mecanico in Mecanico.get_mecanicos():
                    if especialidad == mecanico.get_especialidad() and auto.get_marca() == mecanico.get_autos():
                        mechs.add(mecanico)
                    salir = "si"
            elif input == 8:
                especialidad = "ModificacionFrenos"
                for mecanico in Mecanico.get_mecanicos():
                    if especialidad == mecanico.get_especialidad() and auto.get_marca() == mecanico.get_autos():
                        mechs.add(mecanico)
                    salir = "si"
            elif input == 9:
                especialidad = "ModificacionEscape"
                for mecanico in Mecanico.get_mecanicos():
                    if especialidad == mecanico.get_especialidad() and auto.get_marca() == mecanico.get_autos():
                        mechs.add(mecanico)
                    salir = "si"
            else:
                print("\n¿Salir? (si/no)")
                input()

        return mechs

    @staticmethod
    def mecanicoModif(auto):
        input = 0
        salir = None
        especialidad = None
        mechs =[]

        while salir is None:
            input = int(input())

            if input == 1:
                especialidad = "ModificacionPintura"
                for mecanico in Mecanico.get_mecanicos():
                    if especialidad == mecanico.get_especialidad() and auto.get_marca() == mecanico.get_autos():
                        mechs.add(mecanico)
                    salir = "si"
            elif input == 2:
                especialidad = "ModificacionLlantas"
                for mecanico in Mecanico.get_mecanicos():
                    if especialidad == mecanico.get_especialidad() and auto.get_marca() == mecanico.get_autos():
                        mechs.add(mecanico)
                    salir = "si"
            elif input == 3:
                especialidad = "ModificacionSonido"
                for mecanico in Mecanico.get_mecanicos():
                    if especialidad == mecanico.get_especialidad() and auto.get_marca() == mecanico.get_autos():
                        mechs.add(mecanico)
                    salir = "si"
            elif input == 4:
                especialidad = "ModificacionFrenos"
                for mecanico in Mecanico.get_mecanicos():
                    if especialidad == mecanico.get_especialidad() and auto.get_marca() == mecanico.get_autos():
                        mechs.add(mecanico)
                    salir = "si"
            elif input == 5:
                especialidad = "ModificacionEscape"
                for mecanico in Mecanico.get_mecanicos():
                    if especialidad == mecanico.get_especialidad() and auto.get_marca() == mecanico.get_autos():
                        mechs.add(mecanico)
                    salir = "si"
            else:
                print("\n¿Salir? (si/no)")
                input()

        return mechs

    @staticmethod
    def get_mecanicoPorCedula(cedula):
        finder = None
        for mecanico in Mecanico.mecanicos:
            if mecanico.get_cedula() == cedula:
                finder = mecanico

        return finder

    @staticmethod
    def readByte():
        num = int(input())
        return num

    
    def get_nombre(self):
        return self.nombre

    def get_cedula(self):
        return self.cedula

    def get_telefono(self):
        return self.telefono

    def get_correo(self):
        return self.correo

    def get_direccion(self):
        return self.direccion

    def set_nombre(self, nombre):
        self.nombre = nombre

    def set_cedula(self, cedula):
        self.cedula = cedula

    def set_telefono(self, telefono):
        self.telefono = telefono

    def set_correo(self, correo):
        self.correo = correo

    def set_direccion(self, direccion):
        self.direccion = direccion


    def calcular_salario(self):
        return self.pagoSvcs + self.getSalario()


    def info(self):
        texto = "Nombre del Mecanico: " + self.get_nombre() + "\n"
        return texto
